Detectors/TOF/calibration/testWorkflow/TOFDCSDataProcessorSpec.h

```diff
@@ -0,0 +1,176 @@
+#include "Framework/Task.h"
+#include "Framework/Logger.h"
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace tof
+{
+
+using DPID = o2::dcs::DataPointIdentifier;
+using DPVAL = o2::dcs::DataPointValue;
+using DPCOM = o2::dcs::DataPointCompositeObject;
+using namespace o2::ccdb;
+using CcdbManager = o2::ccdb::BasicCCDBManager;
+using clbUtils = o2::calibration::Utils;
+
+class TOFDCSDataProcessor : public o2::framework::Task
+{
+ public:
+  void init(o2::framework::InitContext& ic) final
+  {
+
+    std::vector<DPID> vect;
+    bool useCCDBtoConfigure = ic.options().get<bool>("use-ccdb-to-configure");
+    if (useCCDBtoConfigure) {
+      LOG(INFO) << "Configuring via CCDB";
+      std::string ccdbpath = ic.options().get<std::string>("ccdb-path");
+      auto& mgr = CcdbManager::instance();
+      mgr.setURL(ccdbpath);
+      CcdbApi api;
+      api.init(mgr.getURL());
+      long ts = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+      std::unordered_map<DPID, std::string>* dpid2DataDesc = mgr.getForTimeStamp<std::unordered_map<DPID, std::string>>("TOF/DCSconfig", ts);
+      for (auto& i : *dpid2DataDesc) {
+        vect.push_back(i.first);
+      }
+    } else {
+      LOG(INFO) << "Configuring via hardcoded strings";
+      std::vector<std::string> aliases = {"tof_hv_vp_[00..89]", "tof_hv_vn_[00..89]", "tof_hv_ip_[00..89]", "tof_hv_in_[00..89]"};
+      std::vector<std::string> aliasesInt = {"TOF_FEACSTATUS_[00..71]"};
+      std::vector<std::string> expaliases = o2::dcs::expandAliases(aliases);
+      std::vector<std::string> expaliasesInt = o2::dcs::expandAliases(aliasesInt);
+      for (const auto& i : expaliases) {
+        vect.emplace_back(i, o2::dcs::RAW_DOUBLE);
+      }
+      for (const auto& i : expaliasesInt) {
+        vect.emplace_back(i, o2::dcs::RAW_INT);
+      }
+    }
+
+    LOG(INFO) << "Listing Data Points for TOF:";
+    for (auto& i : vect) {
+      LOG(INFO) << i;
+    }
+
+    mProcessor = std::make_unique<o2::tof::TOFDCSProcessor>();
+    bool useVerboseMode = ic.options().get<bool>("use-verbose-mode");
+    LOG(INFO) << " ************************* Verbose?" << useVerboseMode;
+    if (useVerboseMode) {
+      mProcessor->useVerboseMode();
+    }
+    mProcessor->init(vect);
+  }
+
+  void run(o2::framework::ProcessingContext& pc) final
+  {
+    auto tfid = o2::header::get<o2::framework::DataProcessingHeader*>(pc.inputs().get("input").header)->startTime;
+    auto dps = pc.inputs().get<gsl::span<DPCOM>>("input");
+    mProcessor->setTF(tfid);
+    mProcessor->process(dps);
+    sendOutput(pc.outputs());
+  }
+
+  void endOfStream(o2::framework::EndOfStreamContext& ec) final
+  {
+    mProcessor->finalize();
+    sendOutput(ec.outputs());
+    const auto& payload = mProcessor->getTOFDPsInfo();
+    auto& info = mProcessor->getccdbDPsInfo();
+    auto image = o2::ccdb::CcdbApi::createObjectImage(&payload, &info);
```

Comment:
so, the double DPs will be saved only once in the very end of the workflow lifetime, which might be large. Is this what you want?

Reply:
For now, I think it is ok, but we can change it to have it ever n TFs. I will do this modification in another PR, if you agree.

Reply:
sure

Reply:
Ok, will take care in January :-)